packages/core/lib/resolve-all-data.ts

```diff
@@ -16,12 +17,13 @@ export async function resolveAllData<
 >(
   data: Partial<Data>,
   config: Config,
+  metadata: MetaData = {},
```

Comment:
Slight bit of nuance needed here. This is technically a breaking change, but `onResolveStart` and `onResolveEnd` should not actually be public (see #899). I'm not sure there's any action on you here, but will think about it.

Reply:
Adding Metadata here was required based on the types but it could be changed from `metadata` to `{}` to discourage use.